scopes/harmony/aspect-loader/plugins.ts

```diff
@@ -116,17 +121,48 @@ export class Plugins {
   /**
    * Get the plugin files from the component.
    */
-  static files(component: Component, defs: PluginDefinition[], resolvePath?: (path: string) => string): string[] {
+
+  static files(
+    component: Component,
+    defs: PluginDefinition[],
+    logger: Logger,
+    resolvePath?: (path: string) => string
+  ): string[] {
     const files = defs.flatMap((pluginDef) => {
-      const matches =
-        typeof pluginDef.pattern === 'string'
-          ? component.filesystem.byGlob([pluginDef.pattern])
-          : component.filesystem.byRegex(pluginDef.pattern);
+      const matches = this.getFileMatches(component, pluginDef);
+
+      const warningMessage = this.constructWarningMessage(component);
+      if (matches.length === 0 && warningMessage && !Plugins.checkedComponents.has(component.id.toString())) {
+        logger.consoleWarning(warningMessage);
+        Plugins.checkedComponents.add(component.id.toString());
+      }
 
       return matches.map((file) => {
         return resolvePath ? resolvePath(file.relative) : file.path;
       });
     });
     return files;
   }
+
+  private static getFileMatches(component: Component, pluginDef: PluginDefinition): any[] {
+    return typeof pluginDef.pattern === 'string'
+      ? component.filesystem.byGlob([pluginDef.pattern])
+      : component.filesystem.byRegex(pluginDef.pattern);
+  }
+
+  private static isImportedComponent(component: Component): boolean {
+    return path.isAbsolute(component.filesystem.files[0]?.path || '');
+  }
+
+  private static constructWarningMessage(component: Component): string | undefined {
```

Comment:
This function name doesn't imply what type of warning it is construct.

Reply:
Yeah, at the beginning, I was constructing two kinds of messages - for installed envs and local, but then I realized that the installed message is wrong since it doesn't load it by pattern only at the bringing (it doesn't have the files for some reason)